src/demo_registrydao/handlers/on_factory_origination.py

```diff
@@ -0,0 +1,24 @@
+from typing import Optional, cast
+
+import demo_registrydao.models as models
+from demo_registrydao.types.registry.storage import RegistryStorage
+from dipdup.config import ContractConfig, StaticTemplateConfig
+from dipdup.models import OperationData, OperationHandlerContext, OriginationContext, TransactionContext
+
+
+async def on_factory_origination(
+    ctx: OperationHandlerContext,
+    registry_origination: OriginationContext[RegistryStorage],
+) -> None:
+    originated_contract = cast(str, registry_origination.data.originated_contract_address)
+    index_name = f'registry_dao_{originated_contract}'
+    if index_name not in ctx.config.indexes:
+        ctx.config.contracts[originated_contract] = ContractConfig(
```

Comment:
Have a feeling that we can hide these internals, smth like
`ctx.add_contract(name=address, address=address, typename='registry')
`ctx.add_index(name='index, template='registry_dao', contract=address, **kwargs)